cmake/macros.cmake

```diff
@@ -1296,12 +1298,13 @@ FUNCTION( FINALIZE_TESTBUILDER )
     ENDIF()
     # If we are using HIP and COMPILE_CXX_AS_HIP is set, change the language
     IF ( USE_HIP AND COMPILE_CXX_AS_HIP )
-        FOREACH ( tmp ${TESTBUILDER_SOURCES} )
+        FOREACH ( tmp ${TB_TARGET_SOURCES} )
             GET_SOURCE_FILE_PROPERTY( lang "${tmp}" LANGUAGE ) 
-            IF ( lang STREQUAL "CXX" )
+            IF ( lang STREQUAL "CXX" OR lang STREQUAL "NOTFOUND" )
```

Comment:
@rbberger , code looks fine, just curious how lang ended up not being found

Reply:
The source files are added in cmake without a file extension. So there is no heuristic CMake can follow to determine the language. The name is further used in the code generation portion without extension.

Reply:
So the alternative is to make that builder "smarter" by accepting files with extension and then stripping it when needed.